webapp/src/pages/EditChatroomDetailsPage/EditChatroomDetailsPage.tsx

```diff
@@ -0,0 +1,217 @@
+        },
+      });
+      notify('Chatroom details successfully updated');
+      navigate(CHAT_URL);
+    } catch (error) {
+      let errMessage = '';
+      if (error instanceof ResponseError) {
+        errMessage = `${error.response.statusText}`;
+      } else if (error instanceof Error) {
+        errMessage = `${error.message}`;
+      }
+      warn(errMessage);
+    }
+  }
+  const handleOnSubmit = (e: React.FormEvent<HTMLFormElement>): void => {
+    e.preventDefault();
+    updateChatroomDetails().catch((e) => console.error(e));
+  };
+  const deleteChatoom = () => {
+    if (window.confirm('Are you sure you want to delete the chatroom?')) {
+      try {
+        chatApi.chatControllerDeleteChatroom({
+          chatroomId: chatroomId!,
+        });
+        notify('Chatroom successfully deleted');
+        navigate(CHAT_URL);
+      } catch (error) {
+        let errMessage = '';
+        if (error instanceof ResponseError) {
+          errMessage = `${error.response.statusText}`;
+        } else if (error instanceof Error) {
+          errMessage = `${error.message}`;
+        }
+        warn(errMessage);
+      }
+    }
+  };
+
+  if (!chatroom) {
+    return (
+      <div className="edit-chatroom-details">
+        <div className="edit-chatroom-details-loading">
+          <Loading />
+        </div>
+      </div>
+    );
+  }
+
+  return (
+    <div className="edit-chatroom-details">
+      <Header icon={IconVariant.ARROW_BACK} onClick={goBack}>
+        {`edit/${chatroom.name}`}
+      </Header>
+      <div className="edit-chatroom-details-avatar-properties">
+        <div className="edit-chatroom-details-avatar">
+          <LargeAvatar
+            url={WILDCARD_AVATAR_URL}
+            editUrl={EDIT_AVATAR_URL}
+            XCoordinate={chatroom.avatarX}
+            YCoordinate={chatroom.avatarY}
+          />
+        </div>
+        <div className="edit-chatroom-details-properties">
+          <ToggleSwitch
+            isToggled={!isPublic}
+            onToggle={() => setIsPublic(!isPublic)}
+            label={isPublic ? 'public channel' : 'private channel'}
+          />
+        </div>
+      </div>
+      <form
+        id="edit-chatroom-details-form"
+        className="edit-chatroom-details-form"
+        onSubmit={handleOnSubmit}
+      >
+        <div className="edit-chatroom-details-form-inputs-container">
+          <Input
+            variant={InputVariant.LIGHT}
+            label="Chat Room Name"
+            placeholder={chatroom.name}
+            value={formValues.name}
+            name="name"
+            onChange={handleInputChange}
+          />
+          <Input
+            variant={InputVariant.LIGHT}
+            label="Old password"
+            placeholder="old password"
+            value={formValues.oldPassword ? formValues.oldPassword : ''}
+            name="oldPassword"
+            type="password"
+            onChange={handleInputChange}
+            disabled={isPublic}
```

Comment:
```suggestion
            disabled={chatroom.isPublic}
```

question: What do you think about using `chatroom.isPublic` here? To check the old password when the user changes the chatroom type from protected to public.

Reply:
Very nice. Done